desktop/ui/src/main/java/org/datacleaner/widgets/DCPersistentSizedPanel.java

```diff
@@ -20,69 +20,33 @@
 package org.datacleaner.widgets;
 
 import java.awt.Dimension;
-import java.util.Map;
 
 import org.datacleaner.panels.DCPanel;
-import org.datacleaner.user.UserPreferences;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
+import org.datacleaner.util.UserPreferencesUtils;
 
 /**
  * A panel that stores it's preferred size in the user preferences.
  * 
- * @author Kasper Sørensen
  */
 public class DCPersistentSizedPanel extends DCPanel {
 
-	private static final long serialVersionUID = 1L;
-	
-	private static final Logger logger = LoggerFactory.getLogger(DCPersistentSizedPanel.class);
-	
-	private final String _identifier;
-	private final int _defaultWidth;
-	private final int _defaultHeight;
-	private final UserPreferences _userPreferences;
+    private static final long serialVersionUID = 1L;
 
-	public DCPersistentSizedPanel(UserPreferences userPreferences, String identifier, int defaultWidth, int defaultHeight) {
-		_identifier = identifier;
-		_defaultWidth = defaultWidth;
-		_defaultHeight = defaultHeight;
-		_userPreferences = userPreferences;
+   // private static final Logger logger = LoggerFactory.getLogger(DCPersistentSizedPanel.class);
 
-		setPreferredSize(getPreferredSizeFromUserPreferences());
-	}
-	
-	@Override
-	public void removeNotify() {
-		super.removeNotify();
-		
-		Map<String, String> properties = _userPreferences.getAdditionalProperties();
+    private final UserPreferencesUtils _userPreferenceUtils; 
 
-		Dimension size = getSize();
-		logger.info("Persisting panel size: {}", size);
-		properties.put(getWidthPropertyKey(), "" + size.width);
-		properties.put(getHeightPropertyKey(), "" + size.height);
-	}
+    public DCPersistentSizedPanel(final UserPreferencesUtils userPreferenceUtils) {
 
-	private Dimension getPreferredSizeFromUserPreferences() {
-		Map<String, String> properties = _userPreferences.getAdditionalProperties();
-		String widthStr = properties.get(getWidthPropertyKey());
-		if (widthStr == null) {
-			widthStr = "" + _defaultWidth;
-		}
-		String heightStr = properties.get(getHeightPropertyKey());
-		if (heightStr == null) {
-			heightStr = "" + _defaultHeight;
-		}
+        _userPreferenceUtils = userPreferenceUtils;
+        final Dimension preferredSizeFromUserPreferences = _userPreferenceUtils.getUserPreferredSize();
+        setPreferredSize(preferredSizeFromUserPreferences);
```

Comment:
Beyond just setting the size we should check if the window was maximized. If it was then we should maximize it (different from setting size).